RELEASE.md

```diff
@@ -1,23 +1,233 @@
+10. Go to the "Common steps" section below.
+
+### Releasing minor version
+
+1. Checkout the correct version branch (e.g. `git checkout "branch-0.14.x"`).
+2. Create some new branch (which you'll later push to your fork).
+3. Perform any backports that are missing from the release.
+3. Create commits that bump version numbers of the crates (see `Version bump commits` section in this document).
+4. Push the branch to your fork and create the PR to the version branch so that another maintainer can review your work.
+Description of this PR should consist of just release notes. Don't merge this PR using GitHub interface - just wait for approval.
+5. Checkout version branch (e.g. `git checkout "branch-0.14.x"`) and fast-forward merge your changes (`git merge --ff-only your_branch`).
+6. Create a new tag (e.g. `git tag -a v0.14.1 -m "Release 0.14.1"`).
+7. Push version branch and new tag, preferably using atomic push (e.g. `git push --atomic origin branch-0.14.x v0.14.1`).
+8. Create a new commit on `main` that updates documentation version - see "Documentation" section in this document. This should build book documentation and publish it on [Documentation book](https://rust-driver.docs.scylladb.com/stable/).
+9. Go to the "Common steps" section below.
+
+
+### Common steps
+
+1. Verify that documentation book was published at [Documentation book](https://rust-driver.docs.scylladb.com/stable/).
+2. Publish all updated crates to crates.io. Make sure you are using newest stable version of Rust. In the main folder of the repository run `cargo publish -p <crate>` for each crate that is being updated, in correct order (`scylla-macros`, `scylla-cql`, `scylla-proxy`, `scylla`).
+3. Verify that new versions are visible on crates.io and their docs on docs.rs.
+4. Go to https://github.com/scylladb/scylla-rust-driver/releases , click the `Draft new release` button and follow the procedure to create a new release on GitHub. Use the release notes as its description.
+5. (Mandatory for major release, optional for minor release) Publish a post on the forum:
+    - Go to [Release notes](https://forum.scylladb.com/c/scylladb-release-notes/18) section.
+    - Click "New Topic".
+    - Title should be `[RELEASE] ScyllaDB Rust Driver <version>`, e.g. `[RELEASE] ScyllaDB Rust Driver 0.14.0`
+    - Tags: `release`, `drivers`, `rust-driver`, `rust`.
+    - Content of the post should just be release notes.
+    - Click "Create Topic"
+    - Posts in "Release notes" section need additional confirmation. You can write to current forum admin to expedite this.
 
 You're done!
 
 
+### Release notes template
+
+```
+The ScyllaDB team is pleased to announce ScyllaDB Rust Driver 0.X.0,
+an asynchronous CQL driver for Rust, optimized for Scylla, but also compatible with Apache Cassandra!
+
+Some interesting statistics:
+
+- over 2.103k downloads on crates!
+- over 556 GitHub stars!
```

Comment:
let's replace exact numbers with some placeholders not to forget updating them when using the template.

Reply:
I left the numbers as-is so that the format of them is easier to understand,

Reply:
So maybe let's add a TODO comment closely?